src/sessions/__tests__/PageManager.test.ts

```diff
@@ -233,6 +233,54 @@ describe('PageManager tests', () => {
         // Assert
         expect(pageManager.getPage()?.interaction).toEqual(1);
 
+        // Assert
+        expect(pageManager.getAttributes()).toMatchObject({
+            pageId: '/console/home',
+            interaction: 1
+        });
+
+        window.removeEventListener(
+            'popstate',
+            (pageManager as any).popstateListener
+        );
+    });
+
+    test('when session resumes and page is manually recorded with custom page attributes then custom page attributes are restored', async () => {
```

Comment:
Let's add a test case where the pagemanager resumes session with page as null - that should be the scenario we need to capture